optuna/integration/allennlp.py

```diff
@@ -0,0 +1,83 @@
+import json
+import os
+
+import optuna
+
+try:
+    import allennlp.commands
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    TrackerCallback = object
+
+
+class AllenNLPExecutor(object):
+    """Allennlp extension to use optuna with an allennlp config file.
+
+    .. warning::
+        AllenNLPExecutor uses environment variables on OS.
+        This could cause problems when AllenNLPExecutor runs in multi-threading.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation
+            of the objective function.
+        config_file:
+            An allennlp config file.
+            Hyperparameters should be masked with `std.extVar`.
+            Please refer to `the config example <https://github.com/allenai/allentune/blob/
+            f2b7de2cad2026c2a50625b939b2db3c1d9bc580/examples/classifier.jsonnet>`_.
```

Comment:
[question] Does allennlp cli things library do some versioning/tagging? If so, I'd refer some specific version.

Reply:
Thank you for catching, it doesn't need to refer the specific version of the example.
I revised my PR to refer to the master branch. 1648abb